src/integer/mat_z/sample/discrete_gauss.rs

```diff
@@ -111,16 +106,41 @@ impl MatZ {
     /// Trapdoors for hard lattices and new cryptographic constructions.
     /// In: Proceedings of the fortieth annual ACM symposium on Theory of computing.
     /// <https://dl.acm.org/doi/pdf/10.1145/1374376.1374407>
-    pub fn sample_d<T1, T2>(basis: &MatZ, n: T1, center: &MatQ, s: T2) -> Result<Self, MathError>
-    where
-        T1: Into<Z>,
-        T2: Into<Q>,
-    {
+    pub fn sample_d(
+        basis: &MatZ,
+        n: impl Into<Z>,
+        center: &MatQ,
+        s: impl Into<Q>,
+    ) -> Result<Self, MathError> {
         let n: Z = n.into();
         let s: Q = s.into();
 
         sample_d(basis, &n, center, &s)
     }
+
+    /// Runs [`MatZ::sample_d`] with identity basis and center vector `0`.
+    /// The full documentation can be found at [`MatZ::sample_d`].
+    ///
+    /// Parameters:
+    /// - `num_rows`: specifies the number of rows that the identity basis should have
+    /// - `num_cols`: specifies the number of columns the identity basis should have
+    /// - `n`: specifies the range from which [`Z::sample_discrete_gauss`] samples
+    /// - `s`: specifies the Gaussian parameter, which is proportional
+    /// to the standard deviation `sigma * sqrt(2 * pi) = s`
+    ///
+    /// Returns a lattice vector sampled according to the discrete Gaussian distribution.
+    /// The lattice specified as `Z^m` for `m = num_rows` and its center fixed to `0^m`.
+    pub fn sample_d_common(
+        num_rows: impl TryInto<i64> + Display,
+        num_cols: impl TryInto<i64> + Display,
+        n: impl Into<Z>,
+        s: impl Into<Q>,
+    ) -> Result<Self, MathError> {
```

Comment:
Add a section "Errors and Failures" and mention the possible error in the Return section